scripts/api_doc_generator.rb

```diff
@@ -0,0 +1,711 @@
+#!/usr/bin/env ruby
```

Comment:
Uuuh do I need to review Ruby? I've never read this language in my life and the $ @ and other sigils scare me.

Reply:
Feel free to skip it. If the generated md files look OK, then this is probably fine.